### PR TITLE
Cyclic groups

### PR DESCRIPTION
**Pull request by EgbertRijke**
In this PR we will show that groups equipped with generating elements are commutative rings.

- Factors out files for commuting elements of semigroups/monoids/groups/rings
- Introduces integer powers of elements of groups
- Introduces integer multiples of elements of abelian groups and rings
- Shows that the evaluation map `hom(G,G) -> G` at a generating element is a group isomorphism
- Refactors isomorphisms of semirings/groups/abelian groups/rings
- Transports ring structures along isomorphisms of abelian groups
- Equips groups with a specified generating element with the structure of a ring
- Shows that the ring structure of a group with specified generating element is commutative

### CODE REVIEW

**Comment by EgbertRijke** [comment]
This one is ready for review!

File: src/group-theory/cyclic-groups.lagda.md [hunk 1]
```diff
@@ -95,11 +145,27 @@ module _
...
+module _
+  {l : Level} (A : Ab l) (a : type-Ab A)
+  where
+
+  map-section-ev-element-hom-Ab : type-Ab A → type-hom-Ab A A
+  map-section-ev-element-hom-Ab x = {! !}
+
+  is-surjective-ev-element-hom-Ab : is-surjective (ev-element-hom-Ab A A a)
+  is-surjective-ev-element-hom-Ab x =
+    unit-trunc-Prop ({!!} , {!!})
```
Comment (EgbertRijke) [inline src/group-theory/cyclic-groups.lagda.md:170]: Ah, I didn't see these open holes.

**Comment by EgbertRijke** [comment]
In the file `integer-powers-of-elements-groups` I tried out the  new style for concatenating equalities. I'm not sure it was an improvement, but I thought it is worth the try. Please let me know what you think.

**Comment by EgbertRijke** [comment]
Now I'm also done with my self-review:)

**Comment by fredrik-bakke** [comment]
I will review this PR now.

File: src/group-theory/integer-powers-of-elements-groups.lagda.md [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+## Idea
+
+The **integer power operation** on a [group](group-theory.groups.md) is the map
+`k x ↦ xᵏ`, which is defined by
+[iteratively](foundation.iterating-automorphisms.md) multiplying `x` with itself
+an [integer](elementary-number-theory.integers.md) `k` times.
+
+## Definitions
+
+### Iterating multiplication by `g`
```
Comment (fredrik-bakke) [inline src/group-theory/integer-powers-of-elements-groups.lagda.md:41]:
```suggestion
### Iterated multiplication by `g`
```

File: src/group-theory/integer-powers-of-elements-groups.lagda.md [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+        ( x ^ inl k)
+        ( commute-inv-Group G (y ^ l) x
+          ( inv (commute-integer-powers-Group' l H)))
+        ( inv (commute-integer-powers-Group (inl k) l H)))
+  commute-integer-powers-Group (inr (inl star)) l {x} {y} H =
+    inv (commute-unit-Group G (y ^ l))
+  commute-integer-powers-Group (inr (inr zero-ℕ)) l {x} {y} H =
+    ap (_* y ^ l) (integer-power-one-Group G x) ∙
+    commute-integer-powers-Group' l H ∙
+    ap (y ^ l *_) (inv (integer-power-one-Group G x))
```
Comment (fredrik-bakke) [inline src/group-theory/integer-powers-of-elements-groups.lagda.md]: I find this notation to be a little confusing, so I suggest parenthesizing the expressions
```suggestion
    ap (_* (y ^ l)) (integer-power-one-Group G x) ∙
    commute-integer-powers-Group' l H ∙
    ap ((y ^ l) *_) (inv (integer-power-one-Group G x))
```

File: src/group-theory/integer-powers-of-elements-groups.lagda.md [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+    (k l : ℤ) {x y : type-Group G} →
+    commute-Group G x y → commute-Group G (x ^ k) (y ^ l)
+  commute-integer-powers-Group (inl zero-ℕ) l {x} {y} H =
+    ap (_* y ^ l) (integer-power-neg-one-Group G x) ∙
+    inv
+      ( commute-inv-Group G
+        ( y ^ l)
+        ( x)
+        ( inv (commute-integer-powers-Group' l H))) ∙
+    ap (y ^ l *_) (inv (integer-power-neg-one-Group G x))
```
Comment (fredrik-bakke) [inline src/group-theory/integer-powers-of-elements-groups.lagda.md]:
```suggestion
    ap ((y ^ l) *_) (inv (integer-power-neg-one-Group G x))
```

File: src/group-theory/integer-powers-of-elements-groups.lagda.md [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+    commute-mul-Group G x y
+      ( integer-power-Group G (inr (inr k)) y)
+      ( H)
+      ( commute-integer-powers-Group' (inr (inr k)) H)
+
+  commute-integer-powers-Group :
+    (k l : ℤ) {x y : type-Group G} →
+    commute-Group G x y → commute-Group G (x ^ k) (y ^ l)
+  commute-integer-powers-Group (inl zero-ℕ) l {x} {y} H =
+    ap (_* y ^ l) (integer-power-neg-one-Group G x) ∙
```
Comment (fredrik-bakke) [inline src/group-theory/integer-powers-of-elements-groups.lagda.md]:
```suggestion
    ap (_* (y ^ l)) (integer-power-neg-one-Group G x) ∙
```

File: src/group-theory/integer-powers-of-elements-groups.lagda.md [hunk 6]
```diff
@@ -0,0 +1,592 @@
...
+    ap (x ^_) (is-right-add-neg-one-pred-ℤ k) ∙
+    distributive-integer-power-add-Group G x k neg-one-ℤ ∙
+    ap (x ^ k *_) (integer-power-neg-one-Group G x)
+
+  integer-power-pred-Group' :
+    (k : ℤ) (x : type-Group G) → x ^ (pred-ℤ k) ＝ x ⁻¹ * x ^ k
+  integer-power-pred-Group' k x =
+    ap (x ^_) (is-left-add-neg-one-pred-ℤ k) ∙
+    distributive-integer-power-add-Group G x neg-one-ℤ k ∙
+    ap (_* x ^ k) (integer-power-neg-one-Group G x)
```
Comment (fredrik-bakke) [inline src/group-theory/integer-powers-of-elements-groups.lagda.md]:
```suggestion
    ap (_* (x ^ k)) (integer-power-neg-one-Group G x)
```

File: src/group-theory/integer-powers-of-elements-groups.lagda.md [hunk 7]
```diff
@@ -0,0 +1,592 @@
...
+
+    infix 55 _⁻¹
+    _⁻¹ = inv-Group G
+
+  integer-power-pred-Group :
+    (k : ℤ) (x : type-Group G) → x ^ (pred-ℤ k) ＝ x ^ k * x ⁻¹
+  integer-power-pred-Group k x =
+    ap (x ^_) (is-right-add-neg-one-pred-ℤ k) ∙
+    distributive-integer-power-add-Group G x k neg-one-ℤ ∙
+    ap (x ^ k *_) (integer-power-neg-one-Group G x)
```
Comment (fredrik-bakke) [inline src/group-theory/integer-powers-of-elements-groups.lagda.md]:
```suggestion
    ap ((x ^ k) *_) (integer-power-neg-one-Group G x)
```

**Comment by fredrik-bakke** [comment]
> In the file `integer-powers-of-elements-groups` I tried out the new style for concatenating equalities. I'm not sure it was an improvement, but I thought it is worth the try. Please let me know what you think.

I think it looks alright. I find it to be a mild improvement. It is probably best to continue to parenthesize each "atomic" factor when they span over multiple lines, though.

**Comment by EgbertRijke** [comment]
One thing that is very good about it, is that the proof term doesn't drift off to the right. I agree about parenthesizing the atomic parts, it will improve readability. I'll go ahead and do that right now.

File: src/commutative-algebra/isomorphisms-commutative-rings.lagda.md [hunk 8]
```diff
@@ -24,22 +33,388 @@ open import ring-theory.isomorphisms-rings
 An **isomorphism** of commutative rings is an invertible homomorphism of
 commutative rings.
```
Comment (fredrik-bakke) [inline src/commutative-algebra/isomorphisms-commutative-rings.lagda.md:34]: add links

File: src/group-theory/commuting-elements-groups.lagda.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+open import group-theory.groups
+```
+
+</details>
+
+## Idea
+
+Two elements `x` and `y` of a [group](group-theory.groups.md) `G` are said to
+**commute** if the equality `xy ＝ yx` holds. For any element `x`, the subset of
+elements that commute with `x` form a subgroup of `G` called the
```
Comment (fredrik-bakke) [inline src/group-theory/commuting-elements-groups.lagda.md:26]:
```suggestion
elements that commute with `x` form a [subgroup](group-theory.subgroups.md) of `G` called the
```

File: src/group-theory/commuting-elements-groups.lagda.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+### If `x` commutes with `y`, then `x * (y * z) ＝ y * (x * z)` for any element `z`
+
+```agda
+module _
+  {l : Level} (G : Group l)
+  where
+
+  private
+
```
Comment (fredrik-bakke) [inline src/group-theory/commuting-elements-groups.lagda.md:109]:
```suggestion
```

File: src/group-theory/commuting-elements-monoids.lagda.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+### If `x` commutes with `y`, then `x * (y * z) ＝ y * (x * z)` for any element `z`
+
+```agda
+module _
+  {l : Level} (M : Monoid l)
+  where
+
+  private
+
```
Comment (fredrik-bakke) [inline src/group-theory/commuting-elements-monoids.lagda.md:92]:
```suggestion
```

File: src/group-theory/commuting-elements-semigroups.lagda.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+### If `x` commutes with `y`, then `x * (y * z) ＝ y * (x * z)` for any element `z`
+
+```agda
+module _
+  {l : Level} (G : Semigroup l)
+  where
+
+  private
+
```
Comment (fredrik-bakke) [inline src/group-theory/commuting-elements-semigroups.lagda.md:83]:
```suggestion
```

File: src/group-theory/commuting-elements-semigroups.lagda.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+### If `x` commutes with `y` and with `z`, then `x` commutes with `yz`
+
+```agda
+module _
+  {l : Level} (G : Semigroup l)
+  where
+
+  private
+
```
Comment (fredrik-bakke) [inline src/group-theory/commuting-elements-semigroups.lagda.md:112]:
```suggestion
```

File: src/group-theory/integer-multiples-of-elements-abelian-groups.lagda.md [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+## Idea
+
+The **integer multiple operation** on an
+[abelian group](group-theory.abelian-groups.md) is the map `k x ↦ kx`, which is
+defined by [iteratively](foundation.iterating-automorphisms.md) adding `x` with
+itself an [integer](elementary-number-theory.integers.md) `k` times.
+
+## Definitions
+
+### Iterating adding by `g`
```
Comment (fredrik-bakke) [inline src/group-theory/integer-multiples-of-elements-abelian-groups.lagda.md:42]:
```suggestion
### Iterated adding by `g`
```

File: src/group-theory/generating-elements-groups.lagda.md [hunk 15]
```diff
@@ -46,10 +61,48 @@ open import group-theory.trivial-group-homomorphisms
 
 An element $g$ of a [group](group-theory.groups.md) `G` is said to be a
 **generating element** if the unique morphism `g̃ : ℤ → G` equipped with an
-identification `g̃(1) = g` is surjective. Equivalently, `g` is a generating
-element if the subgroup `⟨g⟩` of `G`
+identification `g̃(1) = g` is [surjective](foundation.surjective-maps.md).
```
Comment (fredrik-bakke) [inline src/group-theory/generating-elements-groups.lagda.md:64]:
```suggestion
[identification](foundation-core.identity-types.md) `g̃(1) = g` is [surjective](foundation.surjective-maps.md).
```

File: src/group-theory/generating-elements-groups.lagda.md [hunk 16]
```diff
@@ -46,10 +61,48 @@ open import group-theory.trivial-group-homomorphisms
...
+3. The group homomorphism `g̃ : ℤ → G` mapping `1` to `g` is surjective.
+4. The evaluation map `hom(G,H) → H` at `g` is an embedding for every group `H`.
+
+Of these four equivalent characterizations, we take the first as our standard
+definition.
+
+We note that the concept of _group equipped with a generating element_ is subtly
+different from the concept of [cyclic group](group-theory.cyclic-groups.md).
+Groups equipped with generating elements have a specified generating element as
+part of their structure, whereas cyclic groups are groups with the property that
```
Comment (fredrik-bakke) [inline src/group-theory/generating-elements-groups.lagda.md:82]:
```suggestion
part of their structure, whereas cyclic groups are groups with the [property](foundation-core.propositions.md) that
```

File: src/group-theory/isomorphisms-abelian-groups.lagda.md [hunk 17]
```diff
@@ -22,115 +22,175 @@ open import foundation.universe-levels
...
-    {l1 : Level} (A B : Ab l1) → Id A B ≃ iso-Ab A B
+    {l1 : Level} (A B : Ab l1) → Id A B ≃ type-iso-Ab A B
   equiv-iso-eq-Ab' A B =
     ( extensionality-Group' (group-Ab A) (group-Ab B)) ∘e
     ( equiv-ap-inclusion-subtype is-abelian-group-Prop {A} {B})
 
 abstract
   is-contr-total-iso-Ab :
-    { l1 : Level} (A : Ab l1) → is-contr (Σ (Ab l1) (iso-Ab A))
+    { l1 : Level} (A : Ab l1) → is-contr (Σ (Ab l1) (type-iso-Ab A))
```
Comment (fredrik-bakke) [inline src/group-theory/isomorphisms-abelian-groups.lagda.md:193]:
```suggestion
    {l : Level} (A : Ab l) → is-contr (Σ (Ab l) (type-iso-Ab A))
```

File: src/group-theory/isomorphisms-abelian-groups.lagda.md [hunk 18]
```diff
@@ -22,115 +22,175 @@ open import foundation.universe-levels
...
+```agda
 iso-eq-Ab :
-  { l1 : Level} (A B : Ab l1) → Id A B → iso-Ab A B
+  { l1 : Level} (A B : Ab l1) → Id A B → type-iso-Ab A B
 iso-eq-Ab A .A refl = id-iso-Ab A
 
 abstract
   equiv-iso-eq-Ab' :
-    {l1 : Level} (A B : Ab l1) → Id A B ≃ iso-Ab A B
+    {l1 : Level} (A B : Ab l1) → Id A B ≃ type-iso-Ab A B
```
Comment (fredrik-bakke) [inline src/group-theory/isomorphisms-abelian-groups.lagda.md:186]:
```suggestion
    {l : Level} (A B : Ab l) → Id A B ≃ type-iso-Ab A B
```

File: src/group-theory/isomorphisms-abelian-groups.lagda.md [hunk 19]
```diff
@@ -145,6 +205,28 @@ is-equiv-iso-eq-Ab A =
     ( iso-eq-Ab A)
 
 eq-iso-Ab :
-  { l1 : Level} (A B : Ab l1) → iso-Ab A B → Id A B
+  { l1 : Level} (A B : Ab l1) → type-iso-Ab A B → Id A B
```
Comment (fredrik-bakke) [inline src/group-theory/isomorphisms-abelian-groups.lagda.md:208]:
```suggestion
  {l : Level} (A B : Ab l) → type-iso-Ab A B → Id A B
```

File: src/ring-theory/commuting-elements-rings.lagda.md [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+
+### If `x` commutes with `y`, then `x * (y * z) ＝ y * (x * z)` for any element `z`
+
+```agda
+module _
+  {l : Level} (R : Ring l)
+  where
+
+  private
+
```
Comment (fredrik-bakke) [inline src/ring-theory/commuting-elements-rings.lagda.md:177]:
```suggestion
```

File: src/ring-theory/homomorphisms-rings.lagda.md [hunk 21]
```diff
@@ -43,11 +43,10 @@ preserves-mul-hom-Ab :
...
   (x y : type-Ring R) →
-  Id
-    ( map-hom-Ab (ab-Ring R) (ab-Ring S) f (mul-Ring R x y))
-    ( mul-Ring S
-      ( map-hom-Ab (ab-Ring R) (ab-Ring S) f x)
-      ( map-hom-Ab (ab-Ring R) (ab-Ring S) f y))
+  map-hom-Ab (ab-Ring R) (ab-Ring S) f (mul-Ring R x y) ＝
+  mul-Ring S
+    ( map-hom-Ab (ab-Ring R) (ab-Ring S) f x)
+    ( map-hom-Ab (ab-Ring R) (ab-Ring S) f y)
```
Comment (fredrik-bakke) [inline src/ring-theory/homomorphisms-rings.lagda.md:49]:
```suggestion
  ( map-hom-Ab (ab-Ring R) (ab-Ring S) f (mul-Ring R x y)) ＝
  ( mul-Ring S
    ( map-hom-Ab (ab-Ring R) (ab-Ring S) f x)
    ( map-hom-Ab (ab-Ring R) (ab-Ring S) f y))
```

File: src/ring-theory/homomorphisms-rings.lagda.md [hunk 22]
```diff
@@ -383,14 +382,13 @@ id-law-ab-Ring R =
...
-      ( ab-Ring T)
-      ( hom-ab-hom-Ring S T g)
-      ( hom-ab-hom-Ring R S f))
+  hom-ab-hom-Ring R T (comp-hom-Ring R S T g f) ＝
+  comp-hom-Ab
+    ( ab-Ring R)
+    ( ab-Ring S)
+    ( ab-Ring T)
+    ( hom-ab-hom-Ring S T g)
+    ( hom-ab-hom-Ring R S f)
```
Comment (fredrik-bakke) [inline src/ring-theory/homomorphisms-rings.lagda.md:391]:
```suggestion
  ( hom-ab-hom-Ring R T (comp-hom-Ring R S T g f)) ＝
  ( comp-hom-Ab
    ( ab-Ring R)
    ( ab-Ring S)
    ( ab-Ring T)
    ( hom-ab-hom-Ring S T g)
    ( hom-ab-hom-Ring R S f))
```

File: src/ring-theory/integer-multiples-of-elements-rings.lagda.md [hunk 23]
```diff
@@ -0,0 +1,459 @@
...
+## Idea
+
+The **integer multiple operation** on a [ring](ring-theory.rings.md) is the map
+`k x ↦ kx`, which is defined by
+[iteratively](foundation.iterating-automorphisms.md) adding `x` with itself an
+[integer](elementary-number-theory.integers.md) `k` times.
+
+## Definitions
+
+### Iterating adding by `g`
```
Comment (fredrik-bakke) [inline src/ring-theory/integer-multiples-of-elements-rings.lagda.md:42]:
```suggestion
### Iterated adding by `g`
```

File: src/ring-theory/isomorphisms-rings.lagda.md [hunk 24]
```diff
@@ -31,416 +33,478 @@ open import group-theory.homomorphisms-abelian-groups
...
+        ( λ f → is-iso-hom-Ab (ab-Ring R) (ab-Ring S) (pr1 f)))) ∘e
+    ( equiv-type-subtype
+      ( is-prop-is-iso-hom-Ring R S)
+      ( λ f →
+        is-prop-is-iso-hom-Ab
+          ( ab-Ring R)
+          ( ab-Ring S)
+          ( hom-ab-hom-Ring R S f))
+      ( is-iso-hom-ab-is-iso-hom-Ring)
+      ( is-iso-hom-ring-is-iso-hom-Ab))
```
Comment (fredrik-bakke) [inline src/ring-theory/isomorphisms-rings.lagda.md:419]: I expect this proof may become a little more readable if you omit association parentheses.

Comment (fredrik-bakke) [reply src/ring-theory/isomorphisms-rings.lagda.md:419]: Note that this will change the order the equivalences are composed in.

File: src/ring-theory/isomorphisms-rings.lagda.md [hunk 25]
```diff
@@ -31,416 +33,478 @@ open import group-theory.homomorphisms-abelian-groups
...
+  is-equiv-iso-eq-Ring :
+    (S : Ring l) → is-equiv (iso-eq-Ring R S)
+  is-equiv-iso-eq-Ring =
+    fundamental-theorem-id
+      ( is-contr-total-iso-Ring)
+      ( iso-eq-Ring R)
+
+  extensionality-Ring : (S : Ring l) → (R ＝ S) ≃ iso-Ring R S
+  pr1 (extensionality-Ring S) = iso-eq-Ring R S
+  pr2 (extensionality-Ring S) = is-equiv-iso-eq-Ring S
```
Comment (fredrik-bakke) [inline src/ring-theory/isomorphisms-rings.lagda.md:506]: So rings form a category? Is this recorded anywhere?

File: src/ring-theory/isomorphisms-rings.lagda.md [hunk 26]
```diff
@@ -31,416 +33,478 @@ open import group-theory.homomorphisms-abelian-groups
...
+  hom-inv-is-iso-hom-Ring : is-iso-hom-Ring → type-hom-Ring S R
+  hom-inv-is-iso-hom-Ring =
+    hom-inv-is-iso-hom-Large-Precategory
+      ( Ring-Large-Precategory)
+      { X = R}
+      { Y = S}
+      ( f)
+
+  is-section-hom-inv-is-iso-hom-Ring :
+    (U : is-iso-hom-Ring) →
```
Comment (fredrik-bakke) [inline src/ring-theory/isomorphisms-rings.lagda.md:72]: Sorry for asking, but what's the motivation for using `U` for isomorphisms?

**Review by fredrik-bakke — APPROVED** [review]
Very nice! I also had fun learning a little bit of constructive group theory reading through your PR :)

Regarding your choice of making private definitions. I think it looks a little ugly, although I don't object to it.

File: src/group-theory/commuting-elements-groups.lagda.md (hunk 10 above)
Comment (EgbertRijke) [reply src/group-theory/commuting-elements-groups.lagda.md:109]: I think our convention is that we have a blank line after opening a module. `private` opens a private anonymous module, hence the blank line.

Comment (fredrik-bakke) [reply src/group-theory/commuting-elements-groups.lagda.md:109]: Oh, I didn't know that. It looks rather superfluous though

File: src/ring-theory/isomorphisms-rings.lagda.md (hunk 26 above)
Comment (EgbertRijke) [reply src/ring-theory/isomorphisms-rings.lagda.md:72]: Ah, in the groups case `H` was taken, so I started using `U`. It looks like that spilled over to here.

File: src/ring-theory/isomorphisms-rings.lagda.md (hunk 25 above)
Comment (EgbertRijke) [reply src/ring-theory/isomorphisms-rings.lagda.md:506]: Nope, this was going to be my breakfast task tomorrow morning:)

File: src/ring-theory/isomorphisms-rings.lagda.md (hunk 24 above)
Comment (EgbertRijke) [reply src/ring-theory/isomorphisms-rings.lagda.md:419]: Yep, I think in this case it is alright!

File: src/ring-theory/homomorphisms-rings.lagda.md (hunk 22 above)
Comment (EgbertRijke) [reply src/ring-theory/homomorphisms-rings.lagda.md:391]: I think that when it's not in a chain, then it is better without the parentheses

File: src/group-theory/integer-powers-of-elements-groups.lagda.md (hunk 4 above)
Comment (EgbertRijke) [reply src/group-theory/integer-powers-of-elements-groups.lagda.md]: Isn't the point of giving exponentiative operations a higher precedence than multiplication that we can omit them? I thought these rules for exponentiation and multiplication are so standard that the intended meaning is very clear.

File: src/group-theory/commuting-elements-groups.lagda.md (hunk 10 above)
Comment (EgbertRijke) [reply src/group-theory/commuting-elements-groups.lagda.md:109]: To me definitions look better with blank lines above and below them, so I didn't think it was superfluous.

But in fact, you're right that in the case of `abstract` we omit that extra blank line.

You decide:)

Comment (fredrik-bakke) [reply src/group-theory/commuting-elements-groups.lagda.md:109]: Alright, haha, then I want them removed :)

File: src/group-theory/integer-powers-of-elements-groups.lagda.md (hunk 4 above)
Comment (fredrik-bakke) [reply src/group-theory/integer-powers-of-elements-groups.lagda.md]: The point of giving higher precedence was to make it _possible_ to omit the parentheses, not to omit them in every instance. I find that in the expression `y ^ l *_` quite a lot is already going on. It's a fancy notation for a partially evaluated function, it uses privately defined exponentiation and multiplication, and on top of that it uses precedence ordering to shorten the expression. This makes it quite hard to parse for a reader I think, and the extra parentheses help a lot in reducing this complexity.

Comment (EgbertRijke) [reply src/group-theory/integer-powers-of-elements-groups.lagda.md]: Alright, I will comply :)

**Comment by EgbertRijke** [comment]
All done! With this PR we are going to hit 250.000 LOC and 1400 files

**Comment by fredrik-bakke** [comment]
Good work! I bet 300.000 LOC will come even quicker :)

**Comment by EgbertRijke** [comment]
Haha do you have 50000 LOC laying around on your computer?:)

**Comment by fredrik-bakke** [comment]
Who knows 😇
